Corona-Warn-App/src/main/java/de/rki/coronawarnapp/appconfig/mapping/SelfReportSubmissionConfigMapper.kt

```diff
@@ -0,0 +1,68 @@
+package de.rki.coronawarnapp.appconfig.mapping
+
+import dagger.Reusable
+import de.rki.coronawarnapp.appconfig.SafetyNetRequirementsContainer
+import de.rki.coronawarnapp.appconfig.SelfReportSubmissionCommonContainer
+
+import de.rki.coronawarnapp.appconfig.SelfReportSubmissionConfig
+import de.rki.coronawarnapp.appconfig.SelfReportSubmissionConfigContainer
+import de.rki.coronawarnapp.server.protocols.internal.v2.AppConfigAndroid
+import de.rki.coronawarnapp.server.protocols.internal.v2.PpddSrsParameters
+import timber.log.Timber
+import java.time.Duration
+import javax.inject.Inject
+
+@Reusable
+class SelfReportSubmissionConfigMapper @Inject constructor() : SelfReportSubmissionConfig.Mapper {
+    override fun map(rawConfig: AppConfigAndroid.ApplicationConfigurationAndroid): SelfReportSubmissionConfig = try {
+        when {
+            !rawConfig.hasSelfReportParameters() -> {
+                Timber.d("No SelfReportParameters -> set to default")
+                SelfReportSubmissionConfigContainer.DEFAULT
+            }
+            else -> rawConfig.selfReportParameters.map()
+        }
+    } catch (e: Exception) {
+        Timber.d(e, "SelfReportSubmissionConfigMapper failed -> returning default")
+        SelfReportSubmissionConfigContainer.DEFAULT
+    }
+
+    fun PpddSrsParameters.PPDDSelfReportSubmissionParametersAndroid.map(): SelfReportSubmissionConfigContainer {
```

Comment:
`Ppdd` vs `PPDD`. Can you choose either lower case or upper case to have consistencty?

Reply:
it is autogenerated from Proto-Buff